### ISSUE TITLE
[Direct Line] Connection slows down and eventually fails.

### ISSUE THREAD

**Issue by paulfitz72** [Microsoft/BotBuilder#3812]
## Bot Info 
* App ID: 7bc67dc9-02af-4ee5-8bab-097be5f57633
* SDK Platform: NodeJS
* SDK Version: 3.11.0
* Active Channels: Skype and Web
* Deployment Environment: Azure App Service

## Issue Description
Our bot is calling direct line and what we see over time is that the connection slows down and eventually we get a connection refused consistently. The call that is failing is the call to https://directline.botframework.com/v3/directline/conversations/ 

Once the application has failed it requires a restart to get it working again.

We've used curl from the command line to mimic this behaviour from kudu and get the same intermittent failure of the connection which looking at the inconsistent behavior of the direct-line service of the bot framework, this seems to be an issue from the framework itself. An example of the curl command and the failure is shown below

D:\home>curl -X GET https://directline.botframework.com/v3/directline/conversations/
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:20 --:--:--     0curl: (7) Failed to connect to directline.botframework.com port 443: Timed out

**Comment by paulfitz72** [comment]
As an update the bot ran for 7 hours today, when it failed we saw the following in the bot. In the webchat channel we got:

There was an error sending this message to your bot: HTTP status code Forbidden

In the direct line channel we got:

There was an error sending this message to your bot: HTTP status code GatewayTimeout

The calling application is deployed to a WebApp / App Services run time in Azure

**Comment by nwhitmont** [comment]
Hi @paulfitz72 Thanks for reporting the issue. 

Do you have a conversation ID or two, along with the corresponding time that it failed? This will help us pinpoint the time-frame to investigate.

**Comment by nwhitmont** [comment]
@paulfitz72 What is the name of the App Service where the client is deployed? It could be an Azure issue as well.

**Comment by nwhitmont** [comment]
When I checked your bot just now, it was responding normally in the bot portal webchat and there are no errors/warnings present on the Direct Line channel.  

Where are you seeing the errors exactly?

**Comment by paulfitz72** [comment]
The client application name is uk-hr-chatbot-skype-channel-win-prod-webapp (a web app in azure). The failing version makes a websocket connection to directline and it is this that fails after a period of use. To try and isolate the problem we switched in a version of the code that uses http / REST instead of a websocket (this isn't ideal as it isn't as responsive). This was stable so then to prove a point we've deployed the websocket version of the code to BlueMix and are currently testing. This has been stable for a day but we need to leave it a bit longer but this would point to the problem being in the combination of using websockets out of azure hitting some kind of limit.
